pydantic/dataclasses.py

```diff
@@ -107,17 +147,46 @@ def _pydantic_post_init(self: 'DataclassType', *initvars: Any) -> None:
     )
 
     cls.__initialised__ = False
-    cls.__validate__ = classmethod(_validate_dataclass)
-    cls.__get_validators__ = classmethod(_get_validators)
+    cls.__validate__ = classmethod(_validate_dataclass)  # type: ignore[assignment]
+    cls.__get_validators__ = classmethod(_get_validators)  # type: ignore[assignment]
     if post_init_original:
         cls.__post_init_original__ = post_init_original
 
     if cls.__pydantic_model__.__config__.validate_assignment and not frozen:
-        cls.__setattr__ = setattr_validate_assignment
+        cls.__setattr__ = setattr_validate_assignment  # type: ignore[assignment]
 
     return cls
 
 
+@overload
```

Comment:
No biggy, but make these should be in `TYPE_CHECKING`?

Reply:
I tried but doesn't seem to work
And then I saw that it was not in `TYPE_CHECKING` for [root_validator](https://github.com/samuelcolvin/pydantic/blob/master/pydantic/class_validators.py#L99)